[Swift in WebKit] Use swift-format to format _WebKit_SwiftUI code

rr-codes opened the pull request:
#### 0bb7cc8732d6e88c23c7ceee4879b73146619ea1
<pre>
[Swift in WebKit] Use swift-format to format _WebKit_SwiftUI code
<a href="https://bugs.webkit.org/show_bug.cgi?id=293475">https://bugs.webkit.org/show_bug.cgi?id=293475</a>
<a href="https://rdar.apple.com/151908352">rdar://151908352</a>

Reviewed by Abrar Rahman Protyasha.

Having formatted and linted code is paramount to ensuring a maintainable, readable, consistent, and safe codebase.
The best time to have introduced a Swift formatter and linter was before any Swift was introduced in WebKit. But
the second best time is now.

Work towards this goal by adding a `.swift-format` file and formatting the `_WebKit_SwiftUI` code which is a reasonably
small change.

Subsequent PRs will:

1. Lint _WebKit_SwiftUI code
2. Format WebKit code
3. Lint WebKit code
4. Create an automated way to format/lint code and have it integrated with CI

* .swift-format: Added.

Add a configuration file for swift-format. The following formatting options are customized:

- indentConditionalCompilationBlocks: true -&gt; false
- indentation: 2 spaces -&gt; 4 spaces
- lineBreakAroundMultilineExpressionChainComponents: false -&gt; true
- lineBreakBeforeEachArgument: false -&gt; true
- lineBreakBetweenDeclarationAttributes: false -&gt; true
- lineLength: 100 -&gt; 140
- prioritizeKeepingFunctionOutputTogether: false -&gt; true
- tabWidth: 2 -&gt; 4
- spacesBeforeEndOfLineComments: 2 -&gt; 1

All rule options are set to `true` for maximal correctness and safety. If any of these prove to
be too restrictive, they may be reverted to their default values if needed.

* Source/WebKit/_WebKit_SwiftUI/API/View+WebViewModifiers.swift:
* Source/WebKit/_WebKit_SwiftUI/API/WebPage+SwiftUI.swift:
* Source/WebKit/_WebKit_SwiftUI/CrossImportOverlay.swift:
* Source/WebKit/_WebKit_SwiftUI/Implementation/CocoaWebViewAdapter.swift:
* Source/WebKit/_WebKit_SwiftUI/Implementation/Foundation+Extras.swift:
* Source/WebKit/_WebKit_SwiftUI/Implementation/PlatformTextSearching.swift:
* Source/WebKit/_WebKit_SwiftUI/Implementation/SwiftUI+Extras.swift:
* Source/WebKit/_WebKit_SwiftUI/Implementation/ViewModifierContexts.swift:
* Source/WebKit/_WebKit_SwiftUI/Implementation/WebViewRepresentable.swift:

Apply formatting.

Canonical link: <a href="https://commits.webkit.org/295406@main">https://commits.webkit.org/295406@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/a62d1b6db523ba5af49d0a6e686759cae2923c47

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/104996 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/24710 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/15132 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/110213 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/55676 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/25118 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/33254 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/79727 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/108002 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/19548 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/94763 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/60034 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/133/builds/19292 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/55053 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/89015 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/12886 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/112710 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/32161 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/23665 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/88809 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [❌ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/32526 "Failed to checkout and rebase branch from PR 45819") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/90988 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/88437 "Passed tests") | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/22543 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-2-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/11115 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/27511 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17034 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/32085 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/37457 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/31878 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/35219 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/33437 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas reviewed (approved):
The non-default choices look reasonable to me!

rr-codes committed:
[Swift in WebKit] Use swift-format to format _WebKit_SwiftUI code
https://bugs.webkit.org/show_bug.cgi?id=293475
rdar://151908352

Reviewed by Abrar Rahman Protyasha.

Having formatted and linted code is paramount to ensuring a maintainable, readable, consistent, and safe codebase.
The best time to have introduced a Swift formatter and linter was before any Swift was introduced in WebKit. But
the second best time is now.

Work towards this goal by adding a `.swift-format` file and formatting the `_WebKit_SwiftUI` code which is a reasonably
small change.

Subsequent PRs will:

1. Lint _WebKit_SwiftUI code
2. Format WebKit code
3. Lint WebKit code
4. Create an automated way to format/lint code and have it integrated with CI

* .swift-format: Added.

Add a configuration file for swift-format. The following formatting options are customized:

- indentConditionalCompilationBlocks: true -> false
- indentation: 2 spaces -> 4 spaces
- lineBreakAroundMultilineExpressionChainComponents: false -> true
- lineBreakBeforeEachArgument: false -> true
- lineBreakBetweenDeclarationAttributes: false -> true
- lineLength: 100 -> 140
- prioritizeKeepingFunctionOutputTogether: false -> true
- tabWidth: 2 -> 4
- spacesBeforeEndOfLineComments: 2 -> 1

All rule options are set to `true` for maximal correctness and safety. If any of these prove to
be too restrictive, they may be reverted to their default values if needed.

* Source/WebKit/_WebKit_SwiftUI/API/View+WebViewModifiers.swift:
* Source/WebKit/_WebKit_SwiftUI/API/WebPage+SwiftUI.swift:
* Source/WebKit/_WebKit_SwiftUI/CrossImportOverlay.swift:
* Source/WebKit/_WebKit_SwiftUI/Implementation/CocoaWebViewAdapter.swift:
* Source/WebKit/_WebKit_SwiftUI/Implementation/Foundation+Extras.swift:
* Source/WebKit/_WebKit_SwiftUI/Implementation/PlatformTextSearching.swift:
* Source/WebKit/_WebKit_SwiftUI/Implementation/SwiftUI+Extras.swift:
* Source/WebKit/_WebKit_SwiftUI/Implementation/ViewModifierContexts.swift:
* Source/WebKit/_WebKit_SwiftUI/Implementation/WebViewRepresentable.swift:

Apply formatting.

Canonical link: https://commits.webkit.org/295406@main
